components/brave_wallet/browser/json_rpc_service.cc

```diff
@@ -561,19 +557,17 @@ void JsonRpcService::RemoveChain(const std::string& chain_id,
 
 bool JsonRpcService::SetNetwork(const std::string& chain_id,
                                 mojom::CoinType coin,
+                                const absl::optional<url::Origin>& origin,
                                 bool silent) {
-  auto network_url = GetNetworkURL(prefs_, chain_id, coin);
-  if (!network_url.is_valid()) {
+  // We cannot switch to an unknown chain_id
+  if (!KnownChainExists(chain_id, coin) &&
+      !CustomChainExists(prefs_, chain_id, coin)) {
     return false;
   }
```

Comment:
nit: personally I prefer putting this block into SetCurrentChainId to avoid future misuse.

Reply:
fixed in f5a491afdc05ed6049217a61fb9823392968f7a2